src/index.js

```diff
@@ -23,44 +24,48 @@ jsonSchemaAvro.convert = (jsonSchema) => {
     )
   }
   const record = {
-    name: jsonSchemaAvro._idToName(jsonSchema.id || jsonSchema.$id) || 'main',
+    name: jsonSchemaAvro._idToName(jsonSchema, 'main'),
     type: 'record',
     doc: jsonSchema.description,
     fields,
   }
-  const nameSpace = jsonSchemaAvro._idToNameSpace(jsonSchema.id || jsonSchema.$id)
+  const nameSpace = jsonSchemaAvro._idToNameSpace(jsonSchema)
   if (nameSpace) {
     record.namespace = nameSpace
   }
   return record
 }
 
-jsonSchemaAvro._idToNameSpace = (id) => {
+jsonSchemaAvro._idToUrl = (id) => new URL(id, `http://${DEFAULT_NAMESPACE}/`)
+
+jsonSchemaAvro._idToNameSpace = (schema) => {
+  const id = schema.$id || schema.id
   if (!id) {
     return
   }
-  const url = new URL(id, 'http://nonamespace.int/')
+  const url = jsonSchemaAvro._idToUrl(id)
   let nameSpace = []
-  if (url.host !== 'nonamespace.int') {
+  if (url.host !== DEFAULT_NAMESPACE) {
     const reverseHost = url.host.replace(/-/g, '_').split(/\./).reverse()
     nameSpace = nameSpace.concat(reverseHost)
   }
   if (url.pathname) {
+    console.log('got a pathname', url.pathname)
```

Comment:
Did you mean to include this in the 1.8.0 release?

Reply:
no, I'll release a new version shortly.